src/Storage/Controllers/CleanupController.cs

```diff
@@ -0,0 +1,236 @@
+        /// <param name="instanceEventRepository">the instance event repository handler</param>
+        /// <param name="logger">the logger</param>
+        /// <param name="generalSettings">the general settings.</param>
+        public CleanupController(
+            IInstanceRepository instanceRepository,
+            IApplicationRepository applicationRepository,
+            IBlobRepository blobRepository,
+            IDataRepository dataRepository,
+            IInstanceEventRepository instanceEventRepository,
+            ILogger<CleanupController> logger,
+            IOptions<GeneralSettings> generalSettings)
+        {
+            _instanceRepository = instanceRepository;
+            _applicationRepository = applicationRepository;
+            _blobRepository = blobRepository;
+            _dataRepository = dataRepository;
+            _instanceEventRepository = instanceEventRepository;
+            _logger = logger;
+            _usePostgreSQL = generalSettings.Value.UsePostgreSQL;
+        }
+
+        /// <summary>
+        /// Invoke periodic cleanup of instances
+        /// </summary>
+        /// <returns>?</returns>
+        [HttpDelete("cleanupinstances")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ApiExplorerSettings(IgnoreApi = true)]
+        public async Task<ActionResult> CleanupInstances()
+        {
+            if (!_usePostgreSQL)
+            {
+                return Ok();
+            }
+
+            List<Instance> instances = await _instanceRepository.GetHardDeletedInstances();
+            List<string> autoDeleteAppIds = (await _applicationRepository.FindAll())
+                .Where(a => instances.Select(i => i.AppId).ToList().Contains(a.Id) && a.AutoDeleteOnProcessEnd == true)
+                .Select(a => a.Id).ToList();
+
+            Stopwatch stopwatch = Stopwatch.StartNew();
+            int successfullyDeleted = await CleanupInstancesInternal(instances, autoDeleteAppIds);
+            stopwatch.Stop();
+
+            _logger.LogInformation(
+                "NightlyCleanup // Run // {DeleteCount} of {OriginalCount} instances deleted in {Duration} s",
+                successfullyDeleted,
+                instances.Count,
+                stopwatch.Elapsed.TotalSeconds);
+
+            return Ok();
+        }
+
+        /// <summary>
+        /// Invoke periodic cleanup of instances for a specific app
+        /// </summary>
+        /// <returns>?</returns>
+        [HttpDelete("cleanupinstancesforapp/{appId}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ApiExplorerSettings(IgnoreApi = true)]
+        public async Task<ActionResult> CleanupInstancesForApp(string appId)
+        {
+            if (!_usePostgreSQL)
+            {
+                return Ok();
+            }
+
+            appId = appId.Contains('/') ? appId : appId.Replace(appId.Split('-').First() + '-', appId.Split('-').First() + '/');
```

Comment:
I don't think we need to support appId on any format other than (org)/(appname).

Reply:
Will be removed in PR286